packages/@aws-cdk/aws-iam/lib/policy-document.ts

```diff
@@ -144,12 +144,58 @@ export class AccountRootPrincipal extends AccountPrincipal {
 /**
  * A principal representing all identities in all accounts
  */
-export class Anyone extends ArnPrincipal {
+export class AnyPrincipal extends ArnPrincipal {
   constructor() {
     super('*');
   }
 }
 
+/**
+ * A principal representing all identities in all accounts
+ * @deprecated use `AnyPrincipal`
+ */
+export class Anyone extends AnyPrincipal { }
+
+export class CompositePrincipal extends PolicyPrincipal {
+  private readonly principals = new Array<PolicyPrincipal>();
+
+  constructor(principal: PolicyPrincipal, ...additionalPrincipals: PolicyPrincipal[]) {
+    super();
+    this.assumeRoleAction = principal.assumeRoleAction;
+    this.addPrincipals(principal);
+    this.addPrincipals(...additionalPrincipals);
+  }
+
+  public addPrincipals(...principals: PolicyPrincipal[]): this {
+    for (const p of principals) {
+      if (p.assumeRoleAction !== this.assumeRoleAction) {
+        throw new Error(
+          `Cannot add multiple principals with different "assumeRoleAction". ` +
+          `Expecting "${this.assumeRoleAction}", got "${p.assumeRoleAction}"`);
+      }
+      this.principals.push(p);
+    }
+
+    return this;
+  }
+
+  public policyFragment(): PrincipalPolicyFragment {
+    const principalJson: { [key: string]: string[] } = { };
+    const conditions: { [key: string]: any } = { };
+    for (const p of this.principals) {
+      const fragment = p.policyFragment();
+
+      mergePrincipal(principalJson, fragment.principalJson);
+
+      Object.keys(fragment.conditions).map(key => {
+        conditions[key] = fragment.conditions[key];
```

Comment:
This is not the same! 

Multiple principals encode an "OR", whereas multiple conditions encode an "AND". Only way to encode the same OR is to add multiple Statements to the same policy document.

The easiest solution is to require that none of the principals in a `CompositePrincipal` have any conditions attached to them.